provider_shopper/test/catalog_widget_test.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter/material.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/catalog.dart';
+
+Widget createCatalogScreen() => MultiProvider(
+      providers: [
+        Provider(create: (context) => CatalogModel()),
+        ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            cart.catalog = catalog;
+            return cart;
+          },
+        ),
+      ],
+      child: MaterialApp(
+        home: MyCatalog(),
+      ),
+    );
+
+void main() {
+  final catalogListItems = CatalogModel.itemNames;
+
+  group('CatalogScreen Widget Tests', () {
+    testWidgets('Testing the layout and presentation of widgets',
+        (tester) async {
+      await tester.pumpWidget(createCatalogScreen());
+
+      // Verifying the layout
+      expect(find.text('Catalog'), findsOneWidget);
+      expect(find.byIcon(Icons.shopping_cart), findsOneWidget);
+
+      // Testing for the first N(9) items which appear on the screen at first
+      // according to the base viewport
+      for (String item in catalogListItems.sublist(0, 9)) {
+        expect(find.text(item), findsOneWidget);
+      }
+    });
+
+    testWidgets('Testing the scroll behaviour', (tester) async {
```

Comment:
While I admit that it's cool to see scrolling done in one of the widget tests, the behavior that's actually being tested here (and in the next test) is that of the SliverList and SliverAppBar rather that the code that's written specifically for provider_shopper. It just doesn't make sense for the app, and both of these tests should be removed.